lib/ApplicationFeatures/FileSystemFeature.cpp

```diff
@@ -0,0 +1,56 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2022 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Jan Steemann
+////////////////////////////////////////////////////////////////////////////////
+
+#include "FileSystemFeature.h"
+
+#include "ApplicationFeatures/ApplicationServer.h"
+#include "Basics/files.h"
+#include "ProgramOptions/Option.h"
+#include "ProgramOptions/Parameters.h"
+#include "ProgramOptions/ProgramOptions.h"
+
+using namespace arangodb::options;
+
+namespace arangodb {
+
+void FileSystemFeature::collectOptions(
+    std::shared_ptr<ProgramOptions> options) {
+#ifdef __linux__
+  // option is only available on Linux
+  options
+      ->addOption("--use-splice-syscall",
+                  "Use the splice() syscall for file copying (may not be "
+                  "supported on all filesystems",
+                  new BooleanParameter(&_useSplice),
+                  options::makeFlags(Flags::DefaultNoOs, Flags::OsLinux))
```

Comment:
Heretical question: What is the OS flag good for if a pre-processor directive removes the entire code path?
This is mainly a problem for options that Windows and/or macOS only.

Reply:
It is probably still good for tools that, if run on Linux, will otherwise not know that the flag is Linux-specific.

Reply:
If a not-implemented fatal error would be thrown on unsupported OSes, wouldn't that be like the status quo where the server refuses to start if an unknown startup option is passed? (I'm not sure how difficult or cumbersome that would be to implement, but it seems like a sensible middleground)

Reply:
Why would anyone set the option on a non-Linux OS?
It is a _new_ option, so no existing deployment will use it yet, and users need to opt in.
And the documentation clearly states that it is Linux-only.